File: src/scala/Feed.scala
/******************************************************************************

@LICENCE@
 
 *****************************************************************************/
 
package buyat.api.entities

import java.util.{Date}

/**
 * Represents a buy.at feed.
 * @author buy.at
 * @version 1.0
 * @since 1.0
 * @param _feedId Feed ID.
 * @param _feedName Feed name.
 * @param _programmeId ID of the programme the feed belongs to.
 * @param _numberOfProducts The number of products in the feed.
 * @param _lastUpdated The date the feed was last updated.
 */
class Feed(_feedId:Int, _feedName:String, _programmeId:Int, _numberOfProducts:Int, _lastUpdated:Date) extends APIEntity {

  /**
   * Unique ID of the feed.
   */
  var feedId:Int = _feedId

  /**
   * Name of the feed.
   */
  var feedName:String = _feedName

  /**
   * ID of the programme this feed belongs to.
   */
  var programmeId:Int = _programmeId

  /**
   * The number of products in the feed.
   */
  var numberOfProducts:Int = _numberOfProducts

  /**
   * The last date this feed was updated
   */
  var lastUpdated:Date = _lastUpdated
}

File: src/csharp/Feed.cs
/******************************************************************************

@LICENCE@
 
 *****************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;

namespace BuyAtAPI
{

    public class Feed
    {

        private BuyAtAPIClient apiClient;
        private BuyatAffiliateEntitiesFeed soapFeed;
        private Programme programme;

        /// <summary>
        /// Constructor, wrapping SOAP object
        /// </summary>
        public Feed(BuyAtAPIClient apiClient, BuyatAffiliateEntitiesFeed soapFeed)
        {
            this.apiClient = apiClient;
            this.soapFeed = soapFeed;
        }

        /// <summary>
        /// The Feed ID
        /// </summary>
        public int FeedID
        {
            get
            {
                try
                {
                    return Convert.ToInt32(soapFeed.feed.feed_id);
                }
                catch (FormatException)
                {
                    return -1;
                }
            }
        }

        /// <summary>
        /// The Feed Name
        /// </summary>
        public string FeedName
        {
            get
            {
                return soapFeed.feed.feed_name;
            }
        }

        /// <summary>
        /// The Programme ID
        /// </summary>
        public int ProgrammeID
        {
            get
            {
                try
                {
                    return Convert.ToInt32(soapFeed.feed.programme_id);
                }
                catch (FormatException)
                {
                    return -1;
                }
            }
        }

        /// <summary>
        /// The Programme Name
        /// </summary>
        public string ProgrammeName
        {
            get
            {
                return soapFeed.feed.programme_name;
            }
        }

        /// <summary>
        /// The Programme URL
        /// </summary>
        public string ProgrammeURL
        {
            get
            {
                return soapFeed.feed.programme_url;
            }
        }

        /// <summary>
        /// The number of products in the feed
        /// </summary>
        public int NumberOfProducts
        {
            get
            {
                try
                {
                    return Convert.ToInt32(soapFeed.feed.number_of_products);
                }
                catch (FormatException)
                {
                    return -1;
                }
            }
        }

        /// <summary>
        /// The date the feed was last updated
        /// </summary>
        public DateTime LastUpdated
        {
            get
            {
                return soapFeed.feed.last_updated;
            }
        }

        /// <summary>
        /// The programme associated with this feed
        /// </summary>
        public Programme Programme
        {
            get
            {
                if (this.programme == null)
                {
                    this.programme = this.apiClient.GetProgrammeInfo(this.ProgrammeID);
                }
                return this.programme;
            }
        }

    }
}
